readthedocs/projects/forms.py

```diff
@@ -638,8 +646,14 @@ def clean_domain(self):
 
     def clean_canonical(self):
         canonical = self.cleaned_data['canonical']
-        _id = self.initial.get('id')
-        if canonical and Domain.objects.filter(project=self.project, canonical=True).exclude(pk=_id).exists():  # yapf: disabled  # noqa
+        pk = self.instance.pk
+        has_canonical_domain = (
+            Domain.objects
+            .filter(project=self.project, canonical=True)
+            .exclude(pk=pk)
+            .exists()
+        )
+        if canonical and has_canonical_domain:
             raise forms.ValidationError(
                 _('Only 1 Domain can be canonical at a time.'),
```

Comment:
Might as well also fix this error message, domain is not a proper noun.

```suggestion
                _('Only one domain can be canonical at a time.'),
```